core/interfaces/branch.py

```diff
@@ -524,43 +534,36 @@ def get_selected_branch_name(self, selection, line):
         branch_name = segments[1]
         local_region = self.view.get_regions("git_savvy_interface.branch_list")[0]
         if local_region.contains(selection):
-            return branch_name
+            return (None, branch_name)
 
         remotes = self.get_remotes()
         for remote_name in remotes:
             remote_region = self.view.get_regions("git_savvy_interface.branch_list_" + remote_name)
             if remote_region and remote_region[0].contains(selection):
-                return "{}/{}".format(remote_name, branch_name)
-
+                return (remote_name, branch_name)
 
-class GsBranchesFetchAndMergeCommand(TextCommand, GitCommand):
 
+class GsBranchesFetchAndMergeCommand(GsBranchesMergeSelectedCommand):
     """
     Fetch from remote and merge fetched branch into active branch.
     """
 
-    def run(self, edit):
-        sublime.set_timeout_async(self.run_async, 0)
-
     def run_async(self):
         self.interface = ui.get_interface(self.view.id())
-        selection, line = self.interface.get_selection_line()
-        if not line:
-            return
 
-        segments = line.strip("▸ ").split(" ")
-        branch_name = segments[1]
+        branches = self.get_selected_branches()
+        branches_strings = set()
+        for branch in branches:
+            if branch[0] is None:
+                branches_strings.add(branch[1])
 
-        remotes = self.get_remotes()
-        for remote_name in remotes:
-            remote_region = self.view.get_regions("git_savvy_interface.branch_list_" + remote_name)
-            if remote_region and remote_region[0].contains(selection):
-                sublime.status_message("Fetching from `{}`...".format(remote_name))
-                self.fetch(remote=remote_name)
-                self.merge("{}/{}".format(remote_name, branch_name))
+            else:
+                sublime.status_message("Fetching from `{}`...".format(branch[0]))
+                self.fetch(remote=branch[0])
                 sublime.status_message("Fetch and merge complete.")
-                util.view.refresh_gitsavvy(self.view)
-                return
+                branches_strings.add("{}/{}".format(branch[0], branch[1]))
+        self.merge(branches_strings) if not len(branches_strings) == 0 else None
```

Comment:
This still won't work with multiple branches until [the fix mentioned by @randy3k is applied][1].

[1]: https://github.com/divmain/GitSavvy/pull/535#issuecomment-257183051